apps/api/src/app/tenant/usecases/update-tenant/update-tenant.usecase.ts

```diff
@@ -0,0 +1,79 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { TenantRepository } from '@novu/dal';
+import { UpdateTenantCommand } from './update-tenant.command';
+import { TenantEntity } from '@novu/dal';
+
+@Injectable()
+export class UpdateTenant {
+  constructor(private tenantRepository: TenantRepository) {}
+
+  async execute(command: UpdateTenantCommand): Promise<TenantEntity> {
+    const tenant = await this.tenantRepository.findOne({
+      _environmentId: command.environmentId,
+      identifier: command.identifier,
+    });
+
+    if (!tenant) {
+      throw new BadRequestException(
+        `Tenant with identifier: ${command.identifier} is not exists under environment ${command.environmentId}`
+      );
+    }
+
+    const updatePayload: Partial<TenantEntity> = {};
+
+    if (command.name != null) {
+      updatePayload.name = command.name;
+    }
+
+    if (command.data != null) {
+      updatePayload.data = command.data;
+    }
+
+    if (command.newIdentifier != null && command.newIdentifier !== command.identifier) {
+      await this.validateIdentifierDuplication({
```

Comment:
Because it is an identifier that can not be duplicated we throw an error if the user present duplication.

Reply:
you should compare the `indentifier` from the database with a new value from command

Reply:
It should be the same if it was found in the query but would be better anyway. 👍